[YAML] Patch policies: Install new version when app is closed (#49022)
fleetdm/fleet: docs/Configuration/yaml-files.md

A patch policy's `query` automatically updates. Hosts will fail this policy if they’re not running the latest version found in [the app's metadata](https://github.com/fleetdm/fleet/tree/main/ee/maintained-apps/outputs). If `version` is set for `fleet_maintained_apps`, that version is included in the query.

To automatically patch the app when this policy fails, whether or not the app is open, set `install_software` to `true`.

> To automatically patch the app when this policy fails and app is not open, set `patch_when_closed` to `true`. With this option Fleet adds a read-only pre-install query that skips automatic install while the app is open and retries on the next policy run, and it doesn't retry 2 more times. Also, the `continuous_automations_enabled` is automatically set to `true`. 

Reviewer: Might make sense to clarify what happens after those retries hit the maximum. (Does it install even though the app is open, or not install?)

> To automatically patch the app when this policy fails and app is not open, set `patch_when_closed` to `true`. With this option Fleet adds a read-only pre-install query that skips automatic install while the app is open and retries on the next policy run, and it doesn't retry 2 more times. Also, the `continuous_automations_enabled` is automatically set to `true`. 

Author: ```suggestion
To automatically patch the app when this policy fails and app is not open, set `patch_when_closed` to `true`. With this option, Fleet adds a read-only pre-install query that skips automatic install while the app is open and retries on the next policy run. Also, the `continuous_automations_enabled` is automatically set to `true`. 
```

Author: @rachaelshaw I removed this because it's confusing. If `patch_when_closed` is not enabled, then when software install automation is triggered and the install fails, Fleet retries two more times. That adds a bunch of noise, and it's not necessary in cases where the pre-install query shows that the app is open. I don't think we need to document that here. Less words, and less confusion.

To automatically install the app when this policy fails, you can add an automation by setting `install_software` to `true`.

> By default (`only_when_closed: true`), Fleet adds a read-only pre-install condition that skips the automated install while the app is open on the host, and retries the next time the policy runs — this skip doesn't count against the software install retry limit. `fleetctl gitops` rejects a `pre_install_query` set on the referenced `fleet_maintained_apps` entry while `only_when_closed` is `true`, since Fleet manages that query. Set `only_when_closed` to `false` ("Force patch") to install the update the next time the policy fails, whether or not the app is open.

Author: ```suggestion
When `patch_only_when_closed: true`, Fleet adds a read-only pre-install condition that skips the automated install while the app is open on the host, and retries the next time the policy runs. `pre_install_query` set on the referenced `fleet_maintained_apps` entry while `patch_only_when_closed` is `true` is rejected, since Fleet manages that query. Set `patch_only_when_closed` to `false` to install the update the next time the policy fails, whether or not the app is open.
```

> `only_when_closed: true` requires `continuous_automations_enabled: true` on the same policy — `fleetctl gitops` returns an error otherwise.

Author: ```suggestion
When `patch_only_when_closed` is set to `true`, the `continuous_automations_enabled` is automatically set to `true`.```

  type: patch
  fleet_maintained_app_slug: zoom/darwin
  continuous_automations_enabled: true

>   only_when_closed: true # default; update is skipped (and retried later) while Zoom is open

Author: ```suggestion
  patch_only_when_closed: true # default; update is skipped (and retried later) while Zoom is open
```
